src/runtime/methodbinder.cs

```diff
@@ -295,34 +295,53 @@ internal static MethodInfo MatchSignatureAndParameters(MethodInfo[] mi, Type[] g
                             // this logic below handles cases when multiple overloading methods
                             // are ambiguous, hence comparison between Python and CLR types
                             // is necessary
-                            type = null;
-                            
-                            if (_methods.Length>1) { 
-                                IntPtr pyoptype = IntPtr.Zero;
+                            clrtype = null;
+                            IntPtr pyoptype;
+                            if (_methods.Length > 1) {
+                                pyoptype = IntPtr.Zero;
                                 pyoptype = Runtime.PyObject_Type(op);
                                 Exceptions.Clear();
-                                if (pyoptype != IntPtr.Zero) { }
-                                    type = Converter.GetTypeByAlias(pyoptype);
-                                    Runtime.Decref(pyoptype);
+                                if (pyoptype != IntPtr.Zero) {
+                                    clrtype = Converter.GetTypeByAlias(pyoptype);
                                 }
+                                Runtime.Decref(pyoptype);
+                            }
 
 
-                            if (type != null) {
-                                if (pi[n].ParameterType != type) {
-                                    margs = null;
-                                    break;
+                            if (clrtype != null) {
+                                if (pi[n].ParameterType != clrtype) {
+                                    IntPtr pytype = Converter.GetPythonTypeByAlias(pi[n].ParameterType);
+                                    // does pytype require Decref?
```

Comment:
Why is this a question? it's your code, you should understand it. Comments should help the people that come after you to understand the code, not raise questions about its correctness.